### ISSUE TITLE
The id is null message when I try to serialize the @Embeddable class contains "untouched" values

### ISSUE THREAD

**Issue by almothafar** [ebean-orm/ebean#1148]
I'm using play framework, and have quick code to do some quick task, What I'm doing is storing the order and the customer info with his address, then I return back the order as JSON, the problem that I'm getting an exception "Caused by: java.lang.NullPointerException: The id is null"

## Expected behavior
Returning the JSON without any issues even if there is no 'id' for `@Embeddable`.

## Actual behavior
Serialization is not going to work with `@Embedded` field

### Steps to reproduce
For sure there is setters and getters, just removing them to make code smaller for this issue.

```
@Entity
@Table(name = "order_details")
public class OrderDetails  {
    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    @Version
    private long versionNo = 0;

    @Embedded
    private Address shippingAddress;
}

@Embeddable
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class Address {
    @Column
    @Constraints.MaxLength(255)
    @Constraints.Email
    private String customerEmail;

    @Column
    @Constraints.MaxLength(255)
    @JsonInclude(value= JsonInclude.Include.NON_EMPTY)
    private String customerFullName;

    @Column(length = 50)
    @JsonInclude(value= JsonInclude.Include.NON_EMPTY)
    private String customerPhoneNumber;

    @Column(length = 50)
    @JsonInclude(value= JsonInclude.Include.NON_EMPTY)
    private String customerCompanyName;
}
```

Then when I try to do something like:

```
        Address address= new Address ();
        address.setCustomerEmail("email@example.com");
        address.setCustomerFullName("It is me");
        address.setCustomerPhoneNumber(null);
        OrderDetails savedOrder = new OrderDetails();
        savedOrder.setShippingAddress(address);
        savedOrder.insert();
        Json.toJson(savedOrder); // <- EXCEPTION
```

Edit:
I noticed if I set the value with null it is ok, but if I kept it without any change it will cause the exception to happen, as for example, `savedOrder.setShippingAddress(new Address ());` will make the exception on first field `customerEmail`.

I get an exception:
```

2017-10-04 09:57:54 EEST: [error] - handlers.ErrorHandler - @75ghfg0dj: Execution exception in null:null
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:255)
	at play.api.http.HttpErrorHandlerExceptions.throwableToUsefulException(HttpErrorHandler.scala)
	at play.http.DefaultHttpErrorHandler.throwableToUsefulException(DefaultHttpErrorHandler.java:204)
	at play.http.DefaultHttpErrorHandler.onServerError(DefaultHttpErrorHandler.java:164)
	at play.core.j.JavaHttpErrorHandlerAdapter.$anonfun$onServerError$1(JavaHttpErrorHandlerAdapter.scala:22)
	at play.core.j.JavaHelpers.$anonfun$invokeWithContext$1(JavaHelpers.scala:272)
	at play.core.j.JavaHelpers.withContext(JavaHelpers.scala:283)
	at play.core.j.JavaHelpers.withContext$(JavaHelpers.scala:279)
	at play.core.j.JavaHelpers$.withContext(JavaHelpers.scala:292)
	at play.core.j.JavaHelpers.invokeWithContext(JavaHelpers.scala:271)
	at play.core.j.JavaHelpers.invokeWithContext$(JavaHelpers.scala:270)
	at play.core.j.JavaHelpers$.invokeWithContext(JavaHelpers.scala:292)
	at play.core.j.JavaHttpErrorHandlerAdapter.onServerError(JavaHttpErrorHandlerAdapter.scala:22)
	at play.core.server.AkkaHttpServer$$anonfun$$nestedInanonfun$executeHandler$1$1.applyOrElse(AkkaHttpServer.scala:251)
	at play.core.server.AkkaHttpServer$$anonfun$$nestedInanonfun$executeHandler$1$1.applyOrElse(AkkaHttpServer.scala:250)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:412)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at play.api.libs.streams.Execution$trampoline$.execute(Execution.scala:70)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:68)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:284)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:284)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
	at scala.concurrent.Promise.complete(Promise.scala:49)
	at scala.concurrent.Promise.complete$(Promise.scala:48)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:183)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:38)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:81)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:43)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.util.concurrent.CompletionException: java.lang.RuntimeException: java.lang.IllegalArgumentException: The id is null (through reference chain: models.OrderDetails["shippingAddress"]->models.embeddables.Address["customerCompanyName"])
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:593)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:21)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:18)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:63)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:78)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:81)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:55)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:864)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:106)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:103)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:862)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:68)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:284)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:284)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
	at scala.concurrent.Promise.tryFailure(Promise.scala:108)
	at scala.concurrent.Promise.tryFailure$(Promise.scala:108)
	at scala.concurrent.impl.Promise$DefaultPromise.tryFailure(Promise.scala:183)
	at play.api.mvc.Filter$$anon$1$$anonfun$$nestedInanonfun$apply$4$1.applyOrElse(Filters.scala:81)
	at play.api.mvc.Filter$$anon$1$$anonfun$$nestedInanonfun$apply$4$1.applyOrElse(Filters.scala:75)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:412)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
	... 12 more
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: The id is null (through reference chain: models.OrderDetails["shippingAddress"]->models.embeddables.Address["customerCompanyName"])
	at play.libs.Json.toJson(Json.java:77)
	at controllers.orders.OrdersController.addNewOrder(OrdersController.java:88)
	at order.Routes$$anonfun$routes$1.$anonfun$applyOrElse$20(Routes.scala:296)
	at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:137)
	at play.core.routing.HandlerInvokerFactory$JavaActionInvokerFactory$$anon$8$$anon$2$$anon$1.invocation(HandlerInvoker.scala:108)
	at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:76)
	at actions.ActionCreator$1.call(ActionCreator.java:21)
	at actions.TokenSecured.call(TokenSecured.java:44)
	at actions.TokenSecured.call(TokenSecured.java:32)
	at play.core.j.JavaAction.$anonfun$apply$2(JavaAction.scala:111)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:653)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at play.core.j.HttpExecutionContext$$anon$2.run(HttpExecutionContext.scala:56)
	at play.api.libs.streams.Execution$trampoline$.execute(Execution.scala:70)
	at play.core.j.HttpExecutionContext.execute(HttpExecutionContext.scala:48)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:68)
	at scala.concurrent.impl.Promise$KeptPromise$Kept.onComplete(Promise.scala:368)
	at scala.concurrent.impl.Promise$KeptPromise$Kept.onComplete$(Promise.scala:367)
	at scala.concurrent.impl.Promise$KeptPromise$Successful.onComplete(Promise.scala:375)
	at scala.concurrent.impl.Promise.transform(Promise.scala:29)
	at scala.concurrent.impl.Promise.transform$(Promise.scala:27)
	at scala.concurrent.impl.Promise$KeptPromise$Successful.transform(Promise.scala:375)
	at scala.concurrent.Future.map(Future.scala:287)
	at scala.concurrent.Future.map$(Future.scala:287)
	at scala.concurrent.impl.Promise$KeptPromise$Successful.map(Promise.scala:375)
	at scala.concurrent.Future$.apply(Future.scala:653)
	at play.core.j.JavaAction.apply(JavaAction.scala:111)
	at play.api.mvc.Action.$anonfun$apply$2(Action.scala:96)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:302)
	... 13 more
Caused by: java.lang.IllegalArgumentException: The id is null (through reference chain: models.OrderDetails["shippingAddress"]->models.embeddables.Address["customerCompanyName"])
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:2656)
	at play.libs.Json.toJson(Json.java:75)
	... 46 more
Caused by: com.fasterxml.jackson.databind.JsonMappingException: The id is null (through reference chain: models.OrderDetails["shippingAddress"]->models.embeddables.Address["customerCompanyName"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:388)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:348)
	at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:343)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:697)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:155)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:704)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:689)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:155)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:704)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:689)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:155)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:292)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2509)
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:2651)
	... 47 more
Caused by: java.lang.NullPointerException: The id is null
	at io.ebeaninternal.server.querydefn.DefaultOrmQuery.setId(DefaultOrmQuery.java:1486)
	at io.ebeaninternal.server.querydefn.DefaultOrmQuery.setId(DefaultOrmQuery.java:62)
	at io.ebeaninternal.server.core.DefaultBeanLoader.refreshBeanInternal(DefaultBeanLoader.java:301)
	at io.ebeaninternal.server.core.DefaultBeanLoader.loadBean(DefaultBeanLoader.java:240)
	at io.ebeaninternal.server.core.DefaultServer.loadBean(DefaultServer.java:517)
	at io.ebean.bean.EntityBeanIntercept.loadBeanInternal(EntityBeanIntercept.java:812)
	at io.ebean.bean.EntityBeanIntercept.loadBean(EntityBeanIntercept.java:777)
	at io.ebean.bean.EntityBeanIntercept.preGetter(EntityBeanIntercept.java:885)
	at models.embeddables.Address._ebean_get_customerCompanyName(Address.java:6)
	at models.embeddables.Address.getCustomerCompanyName(Address.java:122)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:664)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:689)
	... 57 more

```

**Comment by rbygrave** [comment]
Looks like a duplicate of https://github.com/ebean-orm/ebean/issues/1140 ... which was fixed in 11.1.1 (released last night).  Can you try 11.1.1 ?

**Comment by rbygrave** [comment]
I believe this is fixed in 11.1.1 (as #1140).  I'll close this ... re-open if that isn't the case. thanks.
